src/k8sconfiguration/azext_k8sconfiguration/_params.py

```diff
@@ -0,0 +1,52 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+
+from knack.arguments import CLIArgumentType
+
+from azure.cli.core.commands.parameters import (
+    get_three_state_flag,
+    get_enum_type,
+    tags_type
+)
+
+from azure.cli.core.commands.validators import get_default_location_from_resource_group
+from ._validators import validate_configuration_type
+
+
+def load_arguments(self, _):
+    sourcecontrolconfiguration_type = CLIArgumentType(help='Name of the K8sconfiguration.')
+
+    with self.argument_context('k8sconfiguration') as c:
+        c.argument('tags', tags_type)
+        c.argument('location', validator=get_default_location_from_resource_group)
+        c.argument('name', sourcecontrolconfiguration_type, options_list=['--name', '-n'])
+        c.argument('cluster_name', options_list=['--cluster-name', '-c'], help='Name of the Kubernetes cluster')
+        c.argument('cluster_type', arg_type=get_enum_type(['connectedClusters', 'managedClusters']),
+                   help='Optional parameter to specify Onprem clusters (default) or AKS clusters.')
+        c.argument('repository_url', options_list=['--repository-url', '-u'],
+                   help='Url of the sourceControl repository')
+        c.argument('enable_helm_operator', arg_type=get_three_state_flag(),
+                   help='Enable support for Helm chart deployments')
+        c.argument('cluster_scoped', action='store_true',
+                   help='''Optional switch to set scope of the operator to be cluster. Default scope is 'namespace'.''')
+        c.argument('configuration_type', validator=validate_configuration_type,
+                   arg_type=get_enum_type(['sourceControlConfiguration']),
+                   help='Type of the configuration')
+        c.argument('helm_operator_params',
+                   help='Optional. Chart values for the Helm Operator (if enabled)')
```

Comment:
In our SDK, the arguments do not have [Required] tag.  So, to clarify, I'd want to leave these as is.

Reply:
It's not in SDK. It's in the signature of the function. If there is no default value for the argument, it will marked as [Required] in CLI help message. Take `create_k8sconfiguration` for example:
`
def create_k8sconfiguration(client, resource_group_name, cluster_name, name, repository_url,
                            operator_instance_name=None, operator_namespace='default', cluster_type='connectedClusters',
                            operator_params='', cluster_scoped=None, operator_type='flux', enable_helm_operator=None,
                            helm_operator_version='0.2.0', helm_operator_params=''):
`

`resource_group_name`, `cluster_name`, `name`, `repository_url` here are all marked as [Required], and others are optional because they all have default value.

Reply:
I see, okay, I'll remove them.